controllers/podchaos/containerkill/types.go

```diff
@@ -0,0 +1,188 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package containerkill
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+
+	"time"
+
+	"github.com/go-logr/logr"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/pingcap/chaos-mesh/api/v1alpha1"
+	pb "github.com/pingcap/chaos-mesh/pkg/chaosdaemon/pb"
+	"github.com/pingcap/chaos-mesh/pkg/utils"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	containerKillActionMsg = "delete container"
+)
+
+type Reconciler struct {
+	client.Client
+	Log logr.Logger
+}
+
+func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	var err error
+	now := time.Now()
+
+	r.Log.Info("reconciling container kill")
+	ctx := context.Background()
+
+	var podchaos v1alpha1.PodChaos
+	if err = r.Get(ctx, req.NamespacedName, &podchaos); err != nil {
+		r.Log.Error(err, "unable to get podchaos")
+		return ctrl.Result{}, nil
+	}
+
+	if podchaos.Spec.ContainerName == "" {
+		err = errors.New("ContainerName is empty")
+		r.Log.Error(err, "ContainerName is empty")
```

Comment:
Error string should not be capitalized or end with punctuation.

Reply:
ok. I have modified it.